src/pzprnode.ts

```diff
@@ -15,12 +15,56 @@ const imgdir = process.env.IMG_DIR || './img';
 const hostname = process.env.HTTP_NAME || '127.0.0.1';
 const port = Number(process.env.HTTP_PORT) || 3456;
 
+const globalAny: any = global;
+var samples: any = {};
+
+globalAny.ui = {
+	debug: {
+		addDebugData: function(pid: string, data: { failcheck: any[][] }) {
+			samples[pid] = [];
+			for (var check of data.failcheck) {
+				type Check = {
+					code: string,
+					puzzle: string,
+					flags?: any,
+				};
+				var c: Check = {
+					code: check[0],
+					puzzle: check[1],
+				}
+				if (check.length > 2) {
+					c.flags = check[2];
+				}
+				samples[pid].push(c);
+			}
+		},
+		addRules: function(pid: string, newrules: any) {},
+	},
+}
+
+function loadsample(pid: string): string {
+	var mod = 'pzpr/dist/js/pzpr-samples/' + pid;
+	require(mod);
+	for (var check of samples[pid]) {
```

Comment:
My version iterates backward over the list, while this iterates forward.

One puzzle where this causes different results is Pentominous: https://pzprxs.vercel.app/rules.html?pentominous

I'm not sure if this is a huge deal, so it might be fine to ignore this.

Reply:
I think I'd leave it as is for now -- either option seems a bit arbitrary. If we ever want to do this "right", we could add a `preview` tag to one of the examples.

(I think for pentominous, the best change might be to move that example to the very end? It looks a bit confusing slotted in there on the rules page right now. But yeah minor issue either way.)